backend/app/app/api/api_v1/endpoints/ue_movement.py

```diff
@@ -1,507 +1,318 @@
+        cells = crud.cell.get_multi_by_owner(db=db, owner_id=user.id)
+
+        # Find current position if one exists
+        for index, point in enumerate(points):
+            if (ue.latitude == point.latitude) and (ue.longitude == point.longitude):
+                current_position_index = index
+                break
+
+        handovers[ue.supi] = []
+
+        while True:
+            if supi not in moving_devices:
+                break
+
+            current_position_index = (
+                increment_position(ue.speed) + current_position_index
+            ) % len(points)
+            point = points[current_position_index]
+
+            cell_now, _ = check_distance(point.latitude, point.longitude, cells)
+
+            ue = crud.ue.update_coordinates(
+                db=db, lat=point.latitude, long=point.longitude, db_obj=ue
+            )
+
+            logging.info("The current cell is %d", cell_now)
+            if cell_now and ue.Cell_id != cell_now.id:
+                handovers[ue.supi].append(cell_now.id)
+                ue.Cell_id = cell_now.id
+                crud.ue.update(
+                    db=db,
+                    db_obj=ue,
+                    obj_in={"Cell_id": ue.Cell_id},
+                )
+
+            await location_notification(
+                ue, ue.Cell_id, cell_now.id if cell_now else None
+            )
+
+            await asyncio.sleep(1)
+
+
+async def location_notification(
+    ue: UE, old_cell_id: Optional[int], current_cell_id: Optional[int]
+):
+    db_mongo = client.fastapi
+
+    subscriptions = list(
+        db_mongo["MonitoringEvent"].find(
+            {"supi": str(ue.supi)},
+        ),
+    )
+
+    for doc in subscriptions:
+        sub = doc["subscription"]
+        sub_validate_time = tools.check_expiration_time(
+            expire_time=sub.get("monitorExpireTime")
+        )
+
+        sub_validate_number_of_reports = tools.check_numberOfReports(
+            sub.get("maximumNumberOfReports")
+        )
+
+        if not sub_validate_time or not sub_validate_number_of_reports:
+            crud_mongo.delete_by_uuid(db_mongo, "MonitoringEvent", doc.get("_id"))
+            continue
+
+        if sub.get("maximumNumberOfReports") is not None:
+            sub["maximumNumberOfReports"] = sub["maximumNumberOfReports"] - 1
+            doc["subscription"] = sub
+            crud_mongo.update(db_mongo, "MonitoringEvent", doc.get("_id"), doc)
+
+        monitoringType = sub["monitoringType"]
+
+        if monitoringType == MonitoringType.LOCATION_REPORTING:
+            await handle_location_report_callback(sub, ue)
+
+        elif monitoringType == MonitoringType.LOSS_OF_CONNECTIVITY:
+            asyncio.create_task(
+                handle_loss_connectivity_callback(sub, ue, old_cell_id, current_cell_id)
+            )
+
+        elif monitoringType == MonitoringType.UE_REACHABILITY:
+            await handle_ue_reachability_callback(sub, ue, old_cell_id, current_cell_id)
```

Comment:
All the handlers should use a new task, since we don't want a slow notification destination to stall the movement loop.